frontend/components/common/CompatibleDevice.tsx

```diff
@@ -26,102 +28,134 @@ export function CompatibleDevice({
       maxModelLines > 0 && variants.length > maxModelLines
          ? [variants.slice(0, maxModelLines), variants.slice(maxModelLines)]
          : [variants, []];
-   const { isOpen, getDisclosureProps, getButtonProps, onToggle } =
-      useDisclosure();
+   const [detailsRef, isOpen, toggle] = useDetails();
    return (
-      <NextLink href={device.deviceUrl} passHref>
-         <chakra.a
-            display="flex"
-            alignItems="start"
-            transition="all 300m"
-            {...otherProps}
+      <Box
+         display="flex"
+         alignItems="start"
+         transition="all 300m"
+         {...(!truncateModels && {
+            onClick: (event) => {
+               toggle();
+               event.preventDefault();
+               event.stopPropagation();
+            },
+            onKeyDown: (event) => {
+               if ('Enter' === event.code) {
+                  toggle();
+                  event.preventDefault();
+                  event.stopPropagation();
+               }
+            },
+         })}
+         {...otherProps}
+      >
+         <NextLink href={device.deviceUrl} passHref>
+            <Link>
+               <Img
+                  src={device.imageUrl}
+                  alt={device.deviceName ?? ''}
+                  w="12"
+                  mr="3"
+                  borderWidth="1px"
+                  borderStyle="solid"
+                  borderColor="gray.300"
+                  borderRadius="base"
+               />
+            </Link>
+         </NextLink>
+         <Flex
+            minH="12"
+            flexDir="column"
+            alignSelf="flex-start"
+            justifyContent="center"
          >
-            <Img
-               src={device.imageUrl}
-               alt={device.deviceName ?? ''}
-               w="12"
-               mr="3"
-               borderWidth="1px"
-               borderStyle="solid"
-               borderColor="gray.300"
-               borderRadius="base"
-            />
+            <NextLink href="#compatibility" passHref>
+               <chakra.a>
+                  <Text
+                     my="2px"
+                     fontWeight="medium"
+                     _groupHover={{ color: 'brand.500' }}
+                  >
+                     {device.deviceName}
+                  </Text>
+               </chakra.a>
```

Comment:
If we're going to wrap with chakra.a, is there a reason to also have the Text element? I'd expect chakra.a takes the same style params.

Reply:
I'll try dropping Text, hadn't thought of that.